pkg/umh/topic/builder.go

```
@@ -0,0 +1,449 @@
+		return "", errors.New("name is required")
+	}
+
+	// Calculate total capacity for efficient allocation
+	capacity := len(TopicPrefix) + len(b.level0) + len(b.dataContract) + len(b.name) + 3 // base + 3 dots
+
+	for _, level := range b.locationSublevels {
+		capacity += len(level) + 1 // +1 for dot
+	}
+
+	if b.virtualPath != nil {
+		capacity += len(*b.virtualPath) + 1 // +1 for dot
```

golangci-lint (1.64.8)

ineffectual assignment to capacity

(ineffassign)